File: DesignPatternsExercises/ruby/CommandDesignPattern/csv_extraction.rb
require_relative 'extraction'
require_relative 'csv_extractor'

class CSVExtraction < Extraction

  attr_accessor :file

  def initialize(file)
    @file = file
  end

  def call_correct_extraction
    # _________(6)_________
    CSVExtractor.new.extract
  end

end

File: DesignPatternsExercises/java/CommandDesignPattern/Solution/CsvExtraction.java
// class CsvExtraction implements __________(5)__________ {
class CsvExtraction implements Extraction {
	Extractor csvExtractor = new CsvExtractor();

	@Override
	public void callCorrectExtraction() {
		// csvExtractor.__________(6)__________();
		csvExtractor.extract();
	}
}
